src/main/java/bio/terra/service/job/StairwayLoggingHooks.java

```diff
@@ -15,47 +16,91 @@
         "stepIndex: {}, direction: {}, timestamp: {}";
+    // Id of the flight
+    private static final String FLIGHT_ID_KEY = "flightId";
+    // Class of the flight
+    private static final String FLIGHT_CLASS_KEY = "flightClass";
+    // Class of the flight step
+    private static final String FLIGHT_STEP_CLASS_KEY = "flightStepClass";
+    // Direction of the step (START, DO, UNDO or SWITCH)
+    private static final String FLIGHT_STEP_DIRECTION_KEY = "flightStepDirection";
+    // The step's execution order
+    private static final String FLIGHT_STEP_NUMBER_KEY = "flightStepNumber";
+    // The type of operation primarily being logged (startFlight, startStep, endStep or endFlight)
+    private static final String FLIGHT_OPERATION_KEY = "flightStepOperation";
+    private static final String FLIGHT_OPERATION_START = "startFlight";
+    private static final String FLIGHT_OPERATION_END = "endFlight";
+    private static final String FLIGHT_STEP_OPERATION_START = "startStep";
+    private static final String FLIGHT_STEP_OPERATION_END = "endStep";
+
+
     private static final Logger logger = LoggerFactory.getLogger(StairwayHook.class);
 
-    private PerformanceLogger performanceLogger;
+    private final PerformanceLogger performanceLogger;
 
     public StairwayLoggingHooks(PerformanceLogger performanceLogger) {
         this.performanceLogger = performanceLogger;
-        logger.info("Performance logging " + (performanceLogger.isEnabled() ? "ON" : "OFF"));
+        logger.info("Performance logging {}", performanceLogger.isEnabled() ? "ON" : "OFF");
     }
 
     @Override
     public HookAction startFlight(FlightContext context) {
-        logger.info(FlightLogFormat, "startFlight", context.getFlightClassName(),
+        MDC.put(FLIGHT_ID_KEY, context.getFlightId());
+        MDC.put(FLIGHT_CLASS_KEY, context.getFlightClassName());
+        MDC.put(FLIGHT_OPERATION_KEY, FLIGHT_OPERATION_START);
+        logger.info(FlightLogFormat, FLIGHT_OPERATION_START, context.getFlightClassName(),
             context.getFlightId(), Instant.now().atZone(ZoneId.of("Z")).format(DateTimeFormatter.ISO_INSTANT));
-        performanceLogger.log(context.getFlightId(), context.getFlightClassName(), "startFlight");
+        performanceLogger.log(context.getFlightId(), context.getFlightClassName(), FLIGHT_OPERATION_START);
         return HookAction.CONTINUE;
     }
 
     @Override
     public HookAction startStep(FlightContext context) {
-        logger.info(StepLogFormat, "startStep", context.getFlightClassName(), context.getFlightId(),
+        MDC.put(FLIGHT_ID_KEY, context.getFlightId());
+        MDC.put(FLIGHT_CLASS_KEY, context.getFlightClassName());
+        MDC.put(FLIGHT_STEP_CLASS_KEY, context.getStepClassName());
+        MDC.put(FLIGHT_STEP_DIRECTION_KEY, context.getDirection().toString());
+        MDC.put(FLIGHT_STEP_NUMBER_KEY, Integer.toString(context.getStepIndex()));
```

Comment:
Using `String.valueOf()` is generally better IMO as it doesn't need to change if the type of its argument ever changes
```suggestion
        MDC.put(FLIGHT_STEP_NUMBER_KEY, String.valueOf(context.getStepIndex()));
```
`String.valueOf` could be used on `context.getDirection()` as well if you think that might be `null` someday